src/checkovInstaller.ts

```diff
@@ -1,25 +1,75 @@
+import * as path from 'path';
 import { Logger } from 'winston';
 import { asyncExec } from './utils';
 
-const updateCheckovWithPip3 = async (logger: Logger): Promise<void> => {
+const isCheckovInstalledGlobally = async () => {
+    try {
+        await asyncExec('checkov --version');
+        return true;
+    } catch (err) {
+        return false;
+    }
+};
+
+const installOrUpdateCheckovWithPip3 = async (logger: Logger): Promise<string | null> => {
     try {
         logger.info('Trying to install Checkov using pip3.');
-        await asyncExec('pip3 install -U checkov -i https://pypi.org/simple/');
-        logger.info('Checkov installed successfully using pip3.');
+        await asyncExec('pip3 install -U --user --verbose checkov -i https://pypi.org/simple/');
+        
+        if (await isCheckovInstalledGlobally()) {
+            const checkovPath = 'checkov';
+            logger.info('Checkov installed successfully using pip3.', { checkovPath });
+            return checkovPath;
+        }
+
+        const [pythonUserBaseOutput] = await asyncExec('python3 -c \'import site; print(site.USER_BASE)\'');
+        const checkovPath = path.join(pythonUserBaseOutput.trim(), 'bin', 'checkov');
+        return checkovPath;
     } catch (error) {
         logger.error('Failed to install or update Checkov using pip3. Error:', { error });
-        throw new Error('Failed to install or update Checkov using pip3');
+        return null;
+    }
+};
+
+const installOrUpdateCheckovWithPipenv = async (logger: Logger): Promise<string | null> => {
+    try {
+        logger.info('Trying to install Checkov using pipenv.');
+        await asyncExec('pipenv --python 3 install checkov');
```

Comment:
Did you check it works? 
We tried to add it before and had a permission issue if I remember correctly (Something with a pip file).

Reply:
It worked for me. But it does create a Pipfile in the project dir.
I'm working on a fix for that now.